server/midddleware/metrics.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package middleware
+
+import (
+	"context"
+	"github.com/tigrisdata/tigris/server/metrics"
+	"github.com/uber-go/tally"
+	"google.golang.org/grpc"
+	"strings"
+)
+
+type grpcCallData struct {
+	grpcMethod string
+	grpcType   string
+}
+
+func newGrpcCallData(grpcMethod string, grpcType string) grpcCallData {
+	return grpcCallData{grpcMethod: grpcMethod, grpcType: grpcType}
+}
+
+func (g *grpcCallData) getTags() map[string]string {
+	fullMethodParts := strings.Split(g.grpcMethod, "/")
+	grpcService := fullMethodParts[1]
+	grpcMethod := fullMethodParts[2]
```

Comment:
just use it directly below, no need in new variable

Reply:
fixed